### PR TITLE
AsyncSocketServer sample: delete the AsyncUserToken class

### LINKED ISSUE #3380: Missing AsyncUserToken
For more information, see this [Stackoverflow question](https://stackoverflow.com/questions/58471688).

### PR DESCRIPTION
**Pull request by antonfirsov**
## Summary

There is no value in having a type that only wraps a `Socket`, we acn simplify the example by storing the `Socket` as `e.UserToken` directly.

Fixes #3380

### CODE REVIEW

**Review by gewarren — APPROVED** [review]

File: snippets/csharp/System.Net.Sockets/SocketAsyncEventArgs/Overview/AsyncSocketServer.cs [hunk 1]
```diff
@@ -257,6 +236,7 @@ public void StartAccept(SocketAsyncEventArgs acceptEventArg)
                 willRaiseEvent = listenSocket.AcceptAsync(acceptEventArg);
                 if (!willRaiseEvent)
                 {
+                    Console.WriteLine(">>>>>>>> sync");
```
Comment (rzikm) [inline snippets/csharp/System.Net.Sockets/SocketAsyncEventArgs/Overview/AsyncSocketServer.cs:239]: Is this addition intentional?

**Review by rzikm — APPROVED** [review]
LGTM

File: snippets/csharp/System.Net.Sockets/SocketAsyncEventArgs/Overview/AsyncSocketServer.cs (hunk 1 above)
Comment (antonfirsov) [reply snippets/csharp/System.Net.Sockets/SocketAsyncEventArgs/Overview/AsyncSocketServer.cs:239]: nope, good catch :)
